### PR TITLE
APID-352 - play 2.8 upgrade changes

### PR DESCRIPTION
**Pull request by sivaprakashiv**

### CODE REVIEW

File: app/uk/gov/hmrc/controllers/DocumentationController.scala [hunk 1]
```diff
@@ -17,12 +17,13 @@
...
 import controllers.Assets
+
 import javax.inject.{Inject, Singleton}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
-import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
 
 @Singleton
-class Documentation @Inject()(assets: Assets, cc: MessagesControllerComponents) extends FrontendBaseController {
+class Documentation @Inject()(assets: Assets, cc: MessagesControllerComponents) extends BackendController(cc) {
```
Comment (srinivasalumunagala) [inline app/uk/gov/hmrc/controllers/DocumentationController.scala:26]: Siva, why replacing FrontendBaseController with FrontendBaseController?

Comment (sivaprakashiv) [reply app/uk/gov/hmrc/controllers/DocumentationController.scala:26]: oh you mean, BackendController ? bootstrap play 2.8 libraries were split into frontend and backend specific now and only one lib is allowed to use. This service is a backend one and hence used BackendController to replace the Frontend one.

Comment (srinivasalumunagala) [reply app/uk/gov/hmrc/controllers/DocumentationController.scala:26]: oh sorry, yah i mean BackendController.. ok then fine. thanks

File: build.sbt [hunk 2]
```diff
@@ -41,30 +41,31 @@ lazy val microservice = (project in file("."))
 
 lazy val appDependencies: Seq[ModuleID] = compile ++ test
 
-lazy val hmrcBootstrapPlay26Version = "1.4.0"
-lazy val hmrcTestVersion = "3.9.0-play-26"
+lazy val hmrcBootstrapPlay28Version = "5.4.0"
```
Comment (peteslater-ee) [inline build.sbt]: 5.16.0 is available - any reason for keeping it on an old version?

File: build.sbt [hunk 3]
```diff
@@ -41,30 +41,31 @@ lazy val microservice = (project in file("."))
...
-  "uk.gov.hmrc" %% "hmrctest" % hmrcTestVersion % "test",
+  "org.webjars" %% "webjars-play" % "2.8.8" % "test",
+  "org.scalatest" %% "scalatest" % "3.2.9" % Test,
+  "com.typesafe.play" %% "play-test" % PlayVersion.current % "test",
+  "org.mockito" %% "mockito-scala-scalatest" % "1.16.46" % Test,
   "org.scalaj" %% "scalaj-http" % scalaJVersion % "test",
   "org.scalatestplus.play" %% "scalatestplus-play" % scalatestPlusPlayVersion % "test",
-  "org.mockito" % "mockito-core" % mockitoVersion % "test",
-  "com.typesafe.play" %% "play-test" % PlayVersion.current % "test",
+  "com.vladsch.flexmark" % "flexmark-all" % "0.35.10" % "test",
```
Comment (peteslater-ee) [inline build.sbt]: bootstrap-test-play includes stuff like scalatest and flexmark as transitive dependencies so you shouldn't need them.

https://github.com/hmrc/bootstrap-play/blob/main/project/LibDependencies.scala#L70

File: app/uk/gov/hmrc/controllers/HeaderValidator.scala [hunk 4]
```diff
@@ -43,7 +40,7 @@ trait HeaderValidator extends Results {
   def validateAccept(rules: Option[String] => Boolean): ActionBuilder[Request, AnyContent] = new ActionBuilder[Request, AnyContent] {
     def invokeBlock[A](request: Request[A], block: Request[A] => Future[Result]): Future[Result] = {
       if (rules(request.headers.get(ACCEPT))) block(request)
-      else Future.successful(Status(ErrorAcceptHeaderInvalid.httpStatusCode)(Json.toJson(ErrorAcceptHeaderInvalid)))
+      else Future.successful(Status(ErrorAcceptHeaderInvalid.httpStatusCode))
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/controllers/HeaderValidator.scala]: Is this deliberate to drop the body of the response?

Comment (sivaprakashiv) [reply app/uk/gov/hmrc/controllers/HeaderValidator.scala]: good catch, didn't want to remove

**Review by srinivasalumunagala — APPROVED** [review]

**Pull request merged**